src/file-upload/interfaces.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { BaseComponentProps } from '../internal/base-component';
+import { FormFieldValidationControlProps } from '../internal/context/form-field-context';
+import { NonCancelableEventHandler } from '../internal/events';
+
+export interface FileUploadProps extends BaseComponentProps, FormFieldValidationControlProps {
+  /**
+   * Specifies the native file input `accept` attribute to describe the allow-list of file types.
+   */
+  accept?: string;
+  /**
+   * Specifies whether to add aria-required to the file upload control.
+   */
+  ariaRequired?: boolean;
+  /**
+   * Show file size in the token. Use `i18nStrings.formatFileSize` to customize it.
+   * Default: `false`.
+   */
+  showFileSize?: boolean;
+  /**
+   * Show file last modified timestamp in the token. Use `i18nStrings.formatFileLastModified` to customize it.
+   * Default: `false`.
+   */
+  showFileLastModified?: boolean;
+  /**
+   * Show file thumbnail in the token. Only supported for images.
+   * Default: `false`.
+   */
+  showFileThumbnail?: boolean;
+  /**
+   * Specifies the native file input `multiple` attribute to allow users entering more than one file.
+   * Default: `false`.
+   */
+  multiple?: boolean;
+  /**
+   * Called when the user selects new file(s), or removes a file.
+   * The event `detail` contains the current value of the component.
+   */
+  onChange?: NonCancelableEventHandler<FileUploadProps.ChangeDetail>;
+  /**
+   * Specifies the currently selected file(s).
+   * If you want to clear the selection, use empty array.
+   */
+  value: ReadonlyArray<File>;
+  /**
+   * Specifies the maximum number of displayed file tokens. If the property isn't set, all of the tokens are displayed.
+   */
+  tokenLimit?: number;
+  /**
+   * Constraint text that is displayed below the control. Use this to provide additional information about file size limit, etc.
+   */
+  constraintText?: React.ReactNode;
+  /**
+   * Text that displays as a validation message.
+   */
+  errorText?: React.ReactNode;
+  /**
+   * An array of file errors corresponding to the files in the `value`.
+   */
+  fileErrors?: null | ReadonlyArray<null | string>;
```

Comment:
When would someone want to use `null` instead of `undefined` here?

Reply:
I would use `null` when setting this value explicitly. It can also be set to null by default if used with some form helper library.

Reply:
Do you have a specific example? Because it is 1-way decision, if we find an example in the future, it will be too late, the value is already used.


For example, we use `null` in Select and it is clearly documented how it works: https://github.com/cloudscape-design/components/blob/471735b67df75c4cb69b1b7b201ef05ca6ddf2df/src/select/interfaces.ts#L143-L147

(it cannot use `undefined`, because it causes controllability warnings, but this explanation does not work for file upload)

Reply:
I will remove the `null` option then. It is easy to add it later if requested.